config/src/main/java/com/typesafe/config/impl/SerializedConfigValue.java

```diff
@@ -42,6 +43,12 @@
     // this is the version used by Java serialization, if it increments it's
     // essentially an ABI break and bad
     private static final long serialVersionUID = 1L;
+    private static final int MAX_BYTES_LENGTH = 65535;
+    private static boolean longStr = false; //serves as a signal between writeValueData() and readValueData(). 
```

Comment:
I believe this will not work because serialization and deserialization need not be in the same process or class loader.

Reply:
Yes, I don't see how it could help.
I guess this would be sufficient to always write string length before the string itself. Then the deserialization should read the size before reading the string. Or am I missing something ?

Reply:
You two are absolutely right! Sorry, I misunderstood the format of the output stream. This new commit should be good.

Reply:
Thanks very much for reviewing them!